### PR TITLE
Add manual versioned documentation publishing

### PR DESCRIPTION
**Pull request by nh-exa**
## Description

Add an isolated, manually dispatched foundation for publishing versioned user documentation without coupling it to README migration or the release process.

## Related Issue

[SPOT-32456](https://exasol.atlassian.net/browse/SPOT-32456)

## Type of Change

- [x] `feat`: New feature
- [ ] `fix`: Bug fix
- [x] `docs`: Documentation update
- [ ] `test`: Test additions or changes
- [ ] `refactor`: Code refactoring
- [ ] `chore`: Maintenance tasks

## Changes Made

- Add an isolated `user-docs/` MkDocs and Mike publishing unit with locked dependencies.
- Add locally executable version-catalog management and a thin manual GitHub Pages workflow.
- Document local validation, publication behavior, deletion protection, and recovery.

## Testing

- [x] All existing tests pass (`task all`)
- [ ] Added new tests for new functionality
- [x] Manually tested the changes

### Test Details

- `task fmt`
- `task all` — 119 passed, 6 skipped
- Strict MkDocs build
- Python lint and formatting
- GitHub Actions lint
- Strict OpenSpec validation
- Temporary-repository rehearsal covering stable and prerelease publication, deletion, protected `latest`, and a single explicit push

## Checklist

- [x] Code follows the project's coding guidelines
- [x] Ran `task fmt` and repository linting
- [x] Updated documentation
- [ ] Updated `CHANGELOG.md` — not applicable; no CLI or runtime behavior changed
- [ ] Added/updated automated tests — workflow behavior was validated through an isolated local rehearsal
- [x] All tests pass locally
- [x] Commit messages follow Conventional Commits

## Additional Notes

GitHub Pages is already configured to use GitHub Actions. The real workflow dispatch requires a release-style tag containing this publishing foundation, so end-to-end Pages verification follows after merge.

[SPOT-32456]: https://exasol.atlassian.net/browse/SPOT-32456?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+      target:
+        description: Git tag for publish (v2.3.0-rc1) or version for delete (2.3.0-rc1)
+        required: true
+        type: string
+
+permissions: {}
+
+concurrency:
+  group: documentation-publication
+  cancel-in-progress: false
```
[the remark this answers is not in the document]
Comment (nh-exa) [reply .github/workflows/docs.yml:23]: Addressed in 3421496: the workflow now uses concurrency.queue: max, so publication requests wait in request order instead of replacing an existing pending run.

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+jobs:
+  update-version-catalog:
+    name: Update version catalog
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
```
[the remark this answers is not in the document]
Comment (nh-exa) [reply .github/workflows/docs.yml:32]: Addressed in 3421496: the write-capable catalog job now references the existing github-pages environment. Its repository deployment policy already restricts runs to main.

File: openspec/specs/versioned-documentation-publishing/spec.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
+# versioned-documentation-publishing Specification
+
+## Purpose
+TBD - created by archiving change add-manual-versioned-docs-publishing. Update Purpose after archive.
```
[the remark this answers is not in the document]
Comment (nh-exa) [reply openspec/specs/versioned-documentation-publishing/spec.md]: Addressed in 3960a5a: the canonical capability now has a finalized purpose statement.

File: user-docs/scripts/versions.py [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+import subprocess
+import sys
+
+SEMANTIC_VERSION = re.compile(
+    r"^(0|[1-9][0-9]*)\."
+    r"(0|[1-9][0-9]*)\."
+    r"(0|[1-9][0-9]*)"
+    r"(?:-(?:[0-9A-Za-z]+(?:[.-][0-9A-Za-z]+)*))?"
+    r"(?:\+(?:[0-9A-Za-z]+(?:[.-][0-9A-Za-z]+)*))?$"
+)
```
[the remark this answers is not in the document]
Comment (nh-exa) [reply user-docs/scripts/versions.py:18]: Addressed in df50662: validation now uses the SemVer-compatible identifier pattern, including numeric prerelease leading-zero rules and valid hyphen placement. The edge cases are covered by the new isolated test suite.

**Review by migroble — DISMISSED** [review]

**Review by migroble — APPROVED** [review]
